### PR TITLE
feat: add openGoogleMapsDirections function

### PR DESCRIPTION
**Pull request by gca-axelor**
RM#73971

### CODE REVIEW

**Comment by gca-axelor** [comment]
SettingsScreen
```
/*
 * Axelor Business Solutions
 *
 * Copyright (C) 2024 Axelor (<http://axelor.com>).
 *
 * This program is free software: you can redistribute it and/or  modify
 * it under the terms of the GNU Affero General Public License, version 3,
 * as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React, {useCallback} from 'react';
import {StyleSheet, View} from 'react-native';
import {
  Screen,
  SwitchCard,
  useConfig,
  useTheme,
  Text,
  Button,
} from '@axelor/aos-mobile-ui';
import {useDispatch, useSelector} from 'react-redux';
import {useTranslator} from '../../i18n';
import {
  disable,
  enable,
  useEffectOnline,
  useOnline,
} from '../../features/onlineSlice';
import {ApiProviderConfig} from '../../apiProviders/config';
import {NavigationToolsButton, TranslationsButton} from '../components';
import {linkingProvider} from '../../tools';

const SettingsScreen = ({children}) => {
  const I18n = useTranslator();
  const Theme = useTheme();
  const online = useOnline();
  const dispatch = useDispatch();

  const {
    showFilter,
    hideVirtualKeyboard,
    toggleFilterConfig,
    toggleVirtualKeyboardConfig,
    setShowSubtitles,
    showSubtitles,
  } = useConfig();

  const {appVersion, baseUrl} = useSelector(state => state.auth);

  const handleToggleConnection = useCallback(
    state => {
      if (!state) {
        dispatch(enable());
      } else {
        dispatch(disable());
      }
    },
    [dispatch],
  );

  useEffectOnline();

  const handleToggleColorBlind = useCallback(
    state => {
      if (state) {
        Theme.activateColorBlind();
      } else {
        Theme.desactivateColorBlind();
      }
    },
    [Theme],
  );

  const handleToggleSubtitles = useCallback(
    state => {
      setShowSubtitles(state);
    },
    [setShowSubtitles],
  );

  const pois2 = [
    {address: 'ATL SERVICES, 113 Rue de Bry, 94430 Chennevières-sur-Marne'},
    {
      address:
        'Comptable LGA Conseil Saint-Maur-des-Fossés, 1BIS Av. Foch, 94100 Saint-Maur-des-Fossés',
    },
    {address: '246 rue Paul Vaillant Couturier, 92000 Nanterre'},
    {address: 'LAX Dance Studio, 18bis Villa Riberolle, 75020 Paris'},
    {latitude: 47.163125, longitude: 2.710288},
    {
      address:
        'AAP (Madrassah du Pré-Saint-Gervais), 49 Rue Danton, 93310 Le Pré-Saint-Gervais',
    },
  ];

  return (
    <Screen style={styles.screen}>
      <View style={styles.container}>
        <SwitchCard
          title={I18n.t('User_ShowFilter')}
          defaultValue={showFilter}
          onToggle={toggleFilterConfig}
        />
        <SwitchCard
          title={I18n.t('User_VirtualKeyboardConfig')}
          defaultValue={hideVirtualKeyboard}
          onToggle={toggleVirtualKeyboardConfig}
        />
        <SwitchCard
          title={I18n.t('User_ColorForColorBlind')}
          defaultValue={Theme.isColorBlind}
          onToggle={handleToggleColorBlind}
        />
        {ApiProviderConfig.allowConnectionBlock && (
          <SwitchCard
            title={I18n.t('User_BlockConnection')}
            defaultValue={!online.isEnabled}
            onToggle={handleToggleConnection}
          />
        )}
        <SwitchCard
          title={I18n.t('User_Show_Drawer_Subtitles')}
          defaultValue={showSubtitles}
          onToggle={handleToggleSubtitles}
        />
        {children}
        <TranslationsButton />
        <NavigationToolsButton />
        <Button
          title="test"
          onPress={() => {
            linkingProvider.openGoogleMapsDirections(pois2);
          }}
        />
      </View>
      <View style={styles.footerContainer}>
        <Text>{I18n.t('Base_Version', {appVersion: appVersion})}</Text>
        <Text>{`${I18n.t('Base_ConnectedOn')}:`}</Text>
        <Text numberOfLines={1}>{baseUrl}</Text>
      </View>
    </Screen>
  );
};

const styles = StyleSheet.create({
  screen: {
    justifyContent: 'space-between',
  },
  container: {
    marginTop: 15,
    justifyContent: 'flex-start',
    alignItems: 'center',
    flex: 1,
  },
  footerContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 10,
  },
});

export default SettingsScreen;
```

File: packages/core/src/tools/linking.helper.ts [hunk 1]
```diff
@@ -114,3 +114,18 @@ export const createAndroidNavigaitonURLParams = (
     {param: 'mode', value: transportType},
   ];
 };
+
+export const createGoogleMapsDirectionsURL = (
+  pois: Array<{address: string; latitude?: number; longitude?: number}>,
```
Comment (vhu-axelor) [inline packages/core/src/tools/linking.helper.ts:119]: I think address is not required as user can choose between address and coordinates

File: packages/core/src/tools/LinkingProvider.ts [hunk 2]
```diff
@@ -123,6 +128,17 @@ class LinkingProvider {
...
     });
   }
+
+  async openGoogleMapsDirections(
+    pois: Array<{address: string; latitude?: number; longitude?: number}>,
+  ) {
+    const url = createGoogleMapsDirectionsURL(pois);
+
+    await this.openURL(url, {
+      title: 'Could not open the map app',
```
Comment (vhu-axelor) [inline packages/core/src/tools/LinkingProvider.ts:138]: Maybe precise that you want to open google map. Also when you don't have google map app it opens default web browser, so maybe you need to change the wording

File: packages/core/src/tools/LinkingProvider.ts [hunk 3]
```diff
@@ -123,6 +128,17 @@ class LinkingProvider {
       message: `Error while opening ${address} in the map app`,
     });
   }
+
+  async openGoogleMapsDirections(
+    pois: Array<{address: string; latitude?: number; longitude?: number}>,
```
Comment (vhu-axelor) [inline packages/core/src/tools/LinkingProvider.ts:133]: I think address is not required here too

**Review by vhu-axelor — COMMENTED** [review]
Works well, very nice feature! 🤩
(just few things to change)

**Review by vhu-axelor — APPROVED** [review]

File: packages/core/src/tools/linking.helper.ts [hunk 4]
```diff
@@ -114,3 +114,18 @@ export const createAndroidNavigaitonURLParams = (
...
+  pois: Array<{address?: string; latitude?: number; longitude?: number}>,
+) => {
+  const baseURL = 'https://www.google.com/maps/dir/';
+  const directions = pois
+    .map(poi => {
+      return poi.address
+        ? encodeURI(poi.address)
+        : `${poi.latitude},${poi.longitude}`;
+    })
+    .join('/');
```
Comment (lme-axelor) [inline packages/core/src/tools/linking.helper.ts:128]: Add filter to avoid items when none of the information is given

**Pull request merged**